### PR TITLE
added support for Basic Authentication during symbols loading from sy…

### PR DESCRIPTION
**Pull request by shvez**
…mbols server
As requested in https://github.com/microsoft/perfview/issues/1885 I'm making pull request

### CODE REVIEW

**Comment by pharring** [comment]
@shvez Thanks for the PR. Looks good so far.

A few observations:
I presume you don't want or need any UI for this change? ie There are no scenarios where a user might want to _disable_ the BasicAuth handler?

There is a paragraph or two in the PerfView Users' Guide (UsersGuide.htm) that talks about authentication options for symbols and source. Please add a new entry there describing this new mechanism and showing how to use it.

File: src/PerfView/SymbolReaderHttpHandler.cs [hunk 1]
```diff
@@ -725,6 +728,46 @@ private static string GetAuthenticationHeaderScheme(AuthToken authToken)
...
+        /// <summary>
+        /// Prefix to put in front of logging messages.
+        /// </summary>
+        private const string LogPrefix = "BasicAuth: ";
+
+        public BasicHttpAuthHandler(TextWriter log) : base(log, LogPrefix)
+        {
+        }
+
+
```
Comment (pharring) [inline src/PerfView/SymbolReaderHttpHandler.cs:742]: Remove extra blank line.

File: src/PerfView/SymbolReaderHttpHandler.cs [hunk 2]
```diff
@@ -725,6 +728,46 @@ private static string GetAuthenticationHeaderScheme(AuthToken authToken)
...
+                return base.GetAuthTokenAsync(context, next, authority, cancellationToken);
+            }
+
+            this.WriteStatusLog("auth token for basic auth provided");
+
+            var token = AuthToken.CreateBasicFromUsernameAndPassword(strings[0], strings[1]);
+            return Task.FromResult<AuthToken?>(token);
+        }
+    }
     /// <summary>
```
Comment (pharring) [inline src/PerfView/SymbolReaderHttpHandler.cs:771]: Add missing blank line between closing curly bracket and the start of the doc comment.

File: src/PerfView/MainWindow.xaml [hunk 3]
```diff
@@ -116,6 +117,7 @@
...
                         <MenuItem Header="Use _Developer Identity for Azure DevOps"  x:Name="Authentication_UseManagedIdentityForAzureDevOps" IsChecked="{Binding Path=AuthenticationViewModel.IsDeveloperIdentityEnabled}" Command="{x:Static src:AuthenticationCommands.UseDeveloperIdentity}" ToolTip="Use Visual Studio or VS Code sign-in information to authenticate to Azure DevOps instances."/>
                         <MenuItem Header="Use Device _Code for GitHub" x:Name="Authentication_UseGitHubDeviceFlow" IsChecked="{Binding Path=AuthenticationViewModel.IsGitHubDeviceFlowEnabled}" Command="{x:Static src:AuthenticationCommands.UseGitHubDeviceFlow}" ToolTip="Use a device code to authenticate with GitHub."/>
+                        <MenuItem Header="Use Basic Http Auth" x:Name="Authentication_UseBasicHttpAuth" IsChecked="{Binding Path=AuthenticationViewModel.IsBasicHttpAuthEnabled}" Command="{x:Static src:AuthenticationCommands.UseBasicHttpAuth}" ToolTip="Use a basic http to authenticate on symbol server"/>
```
Comment (pharring) [inline src/PerfView/MainWindow.xaml:120]: For the ToolTip text, I suggest "Use basic HTTP authentication to access a symbol server"

File: src/PerfView/SupportFiles/UsersGuide.htm [hunk 4]
```diff
@@ -8050,6 +8050,12 @@ <h3>
                 and PerfView will be able to access the same private repositories that your account is allowed
                 to access.
             </li>
+            <li>
+                <strong>Basic Http Authentication</strong>. This option allows you to use basic http authentication
+	            when you specify symbol server url. So, full configuration entry may look like this: 
+	            <i>SRV*SymbolCachePath*https://username:password@symbolstore.url;</i>. it is active by default but 
+                used only if url contains <i>username</i> and <i>password</i> information
```
Comment (pharring) [inline src/PerfView/SupportFiles/UsersGuide.htm:8057]: This is pretty good. My suggestion:
```suggestion
                <strong>Basic HTTP Authentication</strong>. This option allows you to use Basic HTTP authentication
	            when connecting to a symbol server. To use it, you should specify the username and password in the URL for your symbol server. For example:
	            <i>SRV*SymbolCachePath*https://username:password@symbolstore.url;</i>. This scheme is active by default but 
                used only if the URL contains <i>username</i> and <i>password</i> information.
```

**Review by pharring — CHANGES_REQUESTED** [review]
Thanks again for working on this. It looks good. Just a few suggestions to fix the wording and whitespace.

**Comment by pharring** [comment]
@brianrob PTAL. This change looks good to me and it addresses #1885

**Comment by brianrob** [comment]
Looks good.  Thank you @shvez and @pharring.

**Comment by pharring** [comment]
Fixes #1885

**Pull request merged**